src/components/CookieProvider.ts

```diff
@@ -0,0 +1,19 @@
+import { Ajax } from 'src/libs/ajax';
+import { AbortOption } from 'src/libs/ajax/data-provider-common';
+import { azureCookieReadyStore, cookieReadyStore } from 'src/libs/state';
+
+export interface CookieProvider {
+  invalidateCookies: (options?: AbortOption) => Promise<void>;
```

Comment:
The signal argument to Ajax is mostly useful for potentially long GET requests where the user might navigate away before the endpoint returns data. I expect that we _never_ want to cancel this request once it's started, so I don't think that `invalidateCookies` needs to accept an abort signal.